hack/testing-olm-upgrade/test-020-olm-upgrade-4.4-4.5.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+# Given an OLM manifest, verify a green field deployment
+# of cluster logging by asserting CLO creates the resources
+# that begets the operands that make up logging.
+
+set -euo pipefail
+
+repo_dir="$( cd "$(dirname "$0")/../.." ; pwd -P )"
+source "$repo_dir/hack/testing-olm/utils"
+source "$repo_dir/hack/testing-olm/assertions"
+
+start_seconds=$(date +%s)
+
+ARTIFACT_DIR=${ARTIFACT_DIR:-"$repo_dir/_output/"}
+ARTIFACT_DIR="${ARTIFACT_DIR}/$(basename ${BASH_SOURCE[0]})"
+if [ ! -d $ARTIFACT_DIR ] ; then
+  mkdir -p $ARTIFACT_DIR
+fi
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+TIMEOUT_MIN=$((2 * $minute))
+ES_POD_TIMEOUT=$((5 * $minute))
+export CLUSTER_LOGGING_OPERATOR_NAMESPACE="openshift-logging"
+manifest_dir=${repo_dir}/manifests
+version="4.5"
+previous_version="4.4"
+
+cleanup(){
+  local return_code="$?"
+  set +e
+  log::info "Running cleanup"
+  end_seconds=$(date +%s)
+  runtime="$(($end_seconds - $start_seconds))s"
+  oc -n openshift-operators-redhat -o yaml get subscription elasticsearch-operator > $ARTIFACT_DIR/subscription-eo.yml 2>&1 ||:
+  oc -n openshift-operators-redhat -o yaml get deployment/elasticsearch-operator > $ARTIFACT_DIR/elasticsearch-operator-deployment.yml 2>&1 ||:
+  oc -n openshift-operators-redhat -o yaml get pods > $ARTIFACT_DIR/openshift-operators-redhat-pods.yml 2>&1 ||:
+  oc -n openshift-operators-redhat -o yaml get configmaps > $ARTIFACT_DIR/openshift-operators-redhat-configmaps.yml 2>&1 ||:
+  oc -n openshift-operators-redhat describe deployment/elasticsearch-operator > $ARTIFACT_DIR/elasticsearch-operator.describe 2>&1 ||:
+
+  oc logs -n "openshift-operators-redhat" deployment/elasticsearch-operator > $ARTIFACT_DIR/elasticsearch-operator.log 2>&1 ||:
+  oc  -n openshift-operator-lifecycle-manager logs --since=$runtime deployment/catalog-operator > $ARTIFACT_DIR/catalog-operator.logs 2>&1 ||:
+  oc  -n openshift-operator-lifecycle-manager logs --since=$runtime deployment/olm-operator > $ARTIFACT_DIR/olm-operator.logs 2>&1 ||:
+
+  ${repo_dir}/olm_deploy/scripts/operator-uninstall.sh
+  ${repo_dir}/olm_deploy/scripts/catalog-uninstall.sh
+
+  set -e
+  exit ${return_code}
+}
+trap cleanup exit
+
+get_es_pods_count() {
+  	oc -n openshift-operators-redhat get pods -l component=elasticsearch --no-headers=true --ignore-not-found | wc -l
+}
+
+get_es_cluster_status() {
+  oc -n openshift-operators-redhat get pods -l component=elasticsearch --no-headers=true --ignore-not-found \
+   | awk 'NR==1{print $1}' \
+   | xargs -I '{}' oc -n openshift-operators-redhat exec '{}' -c elasticsearch -- es_util --query=_cluster/health \
```

Comment:
no need to change but in future note there is a "health" script built into the image